shell.nix

```diff
@@ -1,10 +1,18 @@
 { pkgs ? import <nixpkgs> {} }:
 
+with pkgs;
+
 pkgs.mkShell {
   buildInputs = [
-    pkgs.nodejs-15_x
-
-    # keep this line if you use bash
-    pkgs.bashInteractive
+    nodejs-15_x
+    nodePackages.npm
+    nodePackages.vue-cli
```

Comment:
A good idea, but this won't work. The vue-cli package in the nodePackages references a [depricated npm package](https://www.npmjs.com/package/vue-cli).

You can try to get this working by installing the `@vue/cli` package using [node2nix](https://github.com/svanderburg/node2nix), but I won't fault you for just removing it.